src/plugins/http.js

```diff
@@ -7,66 +7,81 @@ const Tags = opentracing.Tags
 const FORMAT_HTTP_HEADERS = opentracing.FORMAT_HTTP_HEADERS
 
 function patch (http, tracer, config) {
-  this.wrap(http, 'request', request => makeRequestTrace(request))
-  this.wrap(http, 'get', get => makeRequestTrace(get))
+  this.wrap(http, 'request', makeRequestTrace)
+  this.wrap(http, 'get', makeGetTrace)
 
   function makeRequestTrace (request) {
     return function requestTrace (options, callback) {
-      const uri = extractUrl(options)
-      const method = (options.method || 'GET').toUpperCase()
+      const req = request.call(this, options, callback)
 
-      if (uri === `${tracer._url.href}/v0.3/traces`) {
-        return request.apply(this, [options, callback])
+      // The `https.request` function before Node v9 internally invokes `http.request`. When both modules have been
+      // loaded by the user (indirectly) and both modules have been patched by us, that would lead to us double-counting
+      // the request.
+      if (isTraced(req, options)) {
+        return req
+      }
+      // Here we mark the `req` object, as opposed to also marking the `options` object like we do in `get`. The problem
+      // is that if we’d mark `options` here, it would get lost while upstream `request` derives a clone from it and
+      // thus would never get to the recursive invocation of `request`.
```

Comment:
I think the code could be refactored to make the behavior clearer without the need for comments that detailed by using different code paths for the different scenarios. I tried to use the same code for both `http` and `https` but it seems it didn't necessarily make sense.

Reply:
That’s definitely what I _tried_ to do 😅 

Maybe the comments are just making it harder to follow along? Or do you have more concrete suggestions of how to split them up?

Reply:
Mentioned an idea in the `get/request` discussion below, hopefully it makes sense :)